parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java

```diff
@@ -219,6 +222,52 @@ public void testTwoLevelList() throws IOException {
     assertThat(recordRead.get("topbytes")).isEqualTo(expectedBinary);
   }
 
+  @Test
+  public void testParquetRowGroupSize() throws IOException {
+    // verify parquet row group size should be close to configured size
+    int recordCount = 100000;
+    int columnCount = 50;
+
+    List<Types.NestedField> columns =
+        IntStream.rangeClosed(1, columnCount)
+            .mapToObj(i -> optional(i, "stringCol" + i, Types.StringType.get()))
+            .collect(ImmutableList.toImmutableList());
+    Schema schema = new Schema(columns);
+
+    File file = createTempFile(temp);
+
+    List<GenericData.Record> records = Lists.newArrayListWithCapacity(recordCount);
+    org.apache.avro.Schema avroSchema = AvroSchemaUtil.convert(schema.asStruct());
+    for (int i = 1; i <= recordCount; i++) {
+      GenericData.Record record = new GenericData.Record(avroSchema);
+      for (Types.NestedField column : columns) {
+        String value = column.name().repeat(10) + i;
+        record.put(column.name(), value);
+      }
+
+      records.add(record);
+    }
+
+    long actualSize =
+        write(
+            file,
+            schema,
+            ImmutableMap.of("write.parquet.row-group-size-bytes", "1048576"),
+            ParquetAvroWriter::buildWriter,
+            records.toArray(new GenericData.Record[] {}));
+
+    try (ParquetFileReader reader = ParquetFileReader.open(ParquetIO.file(localInput(file)))) {
+      ParquetMetadata footer = reader.getFooter();
+      for (int i = 1; i < footer.getBlocks().size() - 1; i++) {
+        assertThat(footer.getBlocks().get(i).getCompressedSize())
```

Comment:
is it correct to look at the compressed size here instead of looking at `getTotalByteSize()`?

Reply:
If parquet row group size is referring to on disk size, then it should be getCompressedSize(). 
in this test, blocks[i] compressed size is: 1,007,783, total size is: 41,462,625